File: Client_Start/Assets/uLua/Lua/MyLua/Libs/AuxComponent/AuxUIComponent/AuxLabel.lua
require "MyLua.Libs.Core.GlobalNS"
require "MyLua.Libs.Core.Class"
require "MyLua.Libs.Core.GObject"
require "MyLua.Libs.AuxComponent.AuxUIComponent.AuxWindow"

local M = GlobalNS.Class(GlobalNS.AuxWindow);
M.clsName = "AuxLabel";
GlobalNS[M.clsName] = M;

function M:ctor(...)
	--[[
    local params = {...};
    if(GlobalNS.UtilApi.isString(params[2])) then
        self:AuxLabel_1(...);
    else if(type(params[2]) == 'LabelStyleID') then
        self:AuxLabel_2(...);
    else
        self:AuxLabel_3(...);
    end
	]]
end

function M:AuxLabel_1(...)
    local pntNode, path, styleId = ...;
    if(styleId == nil) then
        styleId = LabelStyleID.eLSID_None;
    end
    
    self.m_selfGo = GlobalNS.UtilApi.TransFindChildByPObjAndPath(pntNode, path);
    self.m_text = GlobalNS.UtilApi.getComByP(pntNode, path, 'Text');
    --self.m_labelStyle = Ctx.m_instance.m_widgetStyleMgr.GetWidgetStyle<LabelStyleBase>(WidgetStyleID.eWSID_Text, (int)styleId);
    if(self.m_labelStyle:needClearText()) then
        self.m_text.text = "";
    end
end

function M:AuxLabel_2(...)
    local selfNode, styleId = ...;
    if(styleId == nil) then
        styleId = LabelStyleID.eLSID_None;
    end
    
    self.m_selfGo = selfNode;
    self.m_text = GlobalNS.UtilApi.getComByP(selfNode, 'Text');
end

function M:AuxLabel_3(...)
    local styleId = ...;
    if(styleId == nil) then
        styleId = LabelStyleID.eLSID_None;
    end
end

function M:setSelfGo(pntNode, path)
    m_selfGo = GlobalNS.UtilApi.TransFindChildByPObjAndPath(pntNode, path);
    m_text = GlobalNS.UtilApi.getComByP(pntNode, path, 'Text');
end

function M:setText(value)
    if (self.m_text ~= nil) then
        self.m_text.text = value;
    end
end
    
function M:getText()
    if (self.m_text ~= nil) then
        return self.m_text.text;
    end
    return "";
end

function M:changeSize()
    self.m_text.rectTransform.sizeDelta = Vector2.New(self.m_text.rectTransform.sizeDelta.x, self.m_text.preferredHeight);
end

return M;

File: Client_Start/Assets/Scripts/Libs/AuxComponent/AuxUIComponent/AuxLabel.cs
using UnityEngine;
using UnityEngine.UI;

namespace SDK.Lib
{
    /**
     * @brief 辅助 Label
     */
    public class AuxLabel : AuxWindow
    {
        protected Text mText;
        protected LabelStyleBase mLabelStyle;

        // 初始构造
        public AuxLabel(GameObject pntNode, string path, LabelStyleID styleId = LabelStyleID.eLSID_None)
        {
            this.mSelfGo = UtilApi.TransFindChildByPObjAndPath(pntNode, path);
            this.mText = UtilApi.getComByP<Text>(pntNode, path);
            this.mLabelStyle = Ctx.m_instance.m_widgetStyleMgr.GetWidgetStyle<LabelStyleBase>(WidgetStyleID.eWSID_Text, (int)styleId);

            if(this.mLabelStyle.needClearText())
            {
                this.mText.text = "";
            }
        }

        public AuxLabel(GameObject selfNode, LabelStyleID styleId = LabelStyleID.eLSID_None)
        {
            this.mSelfGo = selfNode;
            this.mText = UtilApi.getComByP<Text>(selfNode);
        }

        public AuxLabel(LabelStyleID styleId = LabelStyleID.eLSID_None)
        {
            
        }

        // 后期修改
        public void setSelfGo(GameObject pntNode, string path)
        {
            this.mSelfGo = UtilApi.TransFindChildByPObjAndPath(pntNode, path);
            this.mText = UtilApi.getComByP<Text>(pntNode, path);
        }

        public string text
        {
            get
            {
                if (this.mText != null)
                {
                    return this.mText.text;
                }
                return "";
            }
            set
            {
                if (this.mText != null)
                {
                    this.mText.text = value;
                }
            }
        }

        public void changeSize()
        {
            this.mText.rectTransform.sizeDelta = new Vector2(this.mText.rectTransform.sizeDelta.x, this.mText.preferredHeight);
        }
    }
}